### PR TITLE
[guiinfo] Add ListItem.SetID infolabel

### PR DESCRIPTION
**Pull request by BigNoid**
Usecase for this is to show a list of movies belonging to the same set as the movie in case `<content target="video">videodb://movies/sets/$INFO[ListItem.SetID]</content>` and to provide a shortcut to open the movie set directly `ActivateWindow(Videos,videodb://movies/sets/$INFO[ListItem.SetID],return)`

@ronie @phil65

### CODE REVIEW

File: xbmc/GUIInfoManager.cpp [hunk 1]
```diff
@@ -5540,6 +5541,15 @@ std::string CGUIInfoManager::GetItemLabel(const CFileItem *item, int info, std::
...
     break;
+  case LISTITEM_SETID:
+    {
+      int iSetId = -1;
+      if (item->HasVideoInfoTag())
+        iSetId = item->GetVideoInfoTag()->m_iSetId;
+      if (iSetId > 0)
+        return StringUtils::Format("%d", iSetId);
+    }
+    break;
```
Comment (phil65) [inline xbmc/GUIInfoManager.cpp]:
```
  case LISTITEM_SETID:
     {
       if (item->HasVideoInfoTag())
         iSetId = item->GetVideoInfoTag()->m_iSetId;
         if (iSetId > 0)
           return StringUtils::Format("%d", iSetId);
     }
     break;

```

Comment (BigNoid) [inline xbmc/GUIInfoManager.cpp]: That wont work as ```int iSetId``` needs to be declared first.

**Comment by phil65** [comment]
oops, sorry.
```
  case LISTITEM_SETID:
     {
       if (item->HasVideoInfoTag())
         int iSetId = item->GetVideoInfoTag()->m_iSetId;
         if (iSetId > 0)
           return StringUtils::Format("%d", iSetId);
     }
     break;

```

**Comment by BigNoid** [comment]
That is not correct i think. I am fairly positive that my code is the correct way of doing this. But seeing that my C++ knowledge is pretty poor, I'll ask @xhaggi to confirm

**Comment by tamland** [comment]
I think what @phil65 tries to say is:

````
if (item->HasVideoInfoTag())
{
  int iSetId = item->GetVideoInfoTag()->m_iSetId;
  if (iSetId > 0)
    return StringUtils::Format("%d", iSetId);
}
````
..but strictly speaking current code should be correct as well.

**Comment by phil65** [comment]
sure, was just cosmetics.

**Comment by xhaggi** [comment]
@tamland you are right ;)
@phil65 it's not just cosmetics because you need to enclose it in brackets if you use more than one line ;)

**Comment by BigNoid** [comment]
I was under the impression that ```iSetId``` always needs to have a value. 
Updated now.

**Comment by phil65** [comment]
hehe youŕe right @xhaggi. I shouldnt write c++ code out of my head without compile-testing. Too  used to other languages. :)
Code looks good now imo.

**Comment by xhaggi** [comment]
looking good now. 

@Razzeee BTW why do we mark it `guilib` if `GUIInfoManager` is not part of our `guilib`? Should we move the GUIInfoManager along with `GUIInfoLabel` into `guiinfo`?

**Comment by Razzeee** [comment]
I just marked it ```guilib``` as @BigNoid used it in the PR name

**Comment by xhaggi** [comment]
I only want to clarify our scopes here. Is `GUIInfoManager` part of our guilib or isn't it?

**Comment by BigNoid** [comment]
It should be ```[guiinfo]``` if I look at history.

**Comment by Razzeee** [comment]
so do you want to change the commit message?

**Comment by BigNoid** [comment]
sure, was just waiting for confirmation.

**Comment by ronie** [comment]
if we're gonna get picky about github labels, then replace the 'guilib' one with 'skin engine'.
that will  cover all i think?

**Comment by Razzeee** [comment]
@ronie 
will do that. 

Anyway I was talking about the actual commit message, somehow missed that @BigNoid already changed it, sorry.

**Comment by ronie** [comment]
@Razzeee make it 'gui engine' please :)

**Pull request merged**